torch/lib/c10d/Utils.hpp

```diff
@@ -203,6 +203,20 @@ inline void assertCPU(
   }
 }
 
+inline void assertSameDevice(
+    std::function<void(const std::string&)> fn,
+    const at::ArrayRef<at::Tensor>& tensors) {
```

Comment:
don't we have a TensorList for this?  (Also I wouldn't expect const reference to it, it's trivial to copy).

Reply:
ah, a good point. I was trying to be consistent with the other functions in this module mostly use const reference to ArrayRef, and not TensorList (TensorList = ArrayRef<Tensor>).
Actually, I just need to verify tensors in a vector, so I might just accept a const ref to a vector.
Would you prefer that?